### PR TITLE
Remove unused fields from serialized state

### LINKED ISSUE #421: Remove Unused Entries in Snippet JSON
## Expected Behavior

There are currently some fields that are saved in the app state which gets saved in S3 that do not need to be there. 

## Current Behavior

Right now, the JSON object that gets saved is of this form:

```javascript
{
  "annotations": { ... },
  "AST": { ... },
  "filters": { ... },
  "hasUnsavedChanges": true/false,
  "snippetLanguage": " ... ",
  "readOnly": true/false,
  "snippet": " ... ",
  "snippetKey": " ... "
  "snippetTitle": " ... ",
}
```

The `hasUnsavedChanges` and `snippetKey` properties do not need to be saved.

Note that this issue should probably be tackled in concert with https://github.com/maryvilledev/codesplain-lambdas/issues/86.

### PR DESCRIPTION
**Pull request by jdmaryville**
## Description

The "snippetKey" and "hasUnsavedChanged" fields will no longer be
included in the serialized JSON object for snippets. They are
stripped out of the state using an axios "transformRequest"
function.

## Motivation and Context

fixes #421 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Browsers

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Mobile:
- [ ] Other:

### CODE REVIEW

File: src/actions/app.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import axios from 'axios';
 
 import { addNotification, closeNotification } from './notifications';
-import { makeSaveEndpointUrl } from '../util/requests';
+import { makeSaveEndpointUrl, stripState } from '../util/requests';
```
Comment (dane-johnson) [inline src/actions/app.js]: Not crazy about this function name.

Comment (Hopding) [inline src/actions/app.js]: What would you prefer?

Comment (solkaz) [inline src/actions/app.js]: I believe most people would refer to this process as "normalizing"

**Comment by thebho** [comment]
Is this good to go?

File: src/actions/app.js (hunk 1 above)
Comment (dane-johnson) [inline src/actions/app.js]: `Normalize` works. Strip is too vague

**Comment by jdmaryville** [comment]
Yes, as well as PR #499.

**Comment by dane-johnson** [comment]
We need another review on both of them

**Pull request merged**